addOns/graphql/src/main/java/org/zaproxy/addon/graphql/ImportDialog.java

```diff
@@ -0,0 +1,245 @@
+                                                ThreadUtils.invokeAndWaitHandled(
+                                                        () -> setVisible(true));
+                                            }
+                                        })
+                                .start();
+                    });
+        }
+        return buttonImport;
+    }
+
+    private JButton getCancelButton() {
+        if (buttonCancel == null) {
+            buttonCancel = new JButton(Constant.messages.getString("all.button.cancel"));
+            buttonCancel.addActionListener(e -> dispose());
+        }
+        return buttonCancel;
+    }
+
+    protected boolean validateEndpointUrl() {
+        try {
+            parser =
+                    new GraphQlParser(
+                            fieldEndpoint.getText(), HttpSender.MANUAL_REQUEST_INITIATOR, false);
+            parser.addRequesterListener(new HistoryPersister());
+            return true;
+        } catch (URIException e) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(
+                                Constant.messages.getString(
+                                        "graphql.error.invalidurl", e.getMessage()));
+                        fieldEndpoint.requestFocusInWindow();
+                    });
+        }
+        return false;
+    }
+
+    private JButton getChooseFileButton() {
+        if (buttonChooseFile == null) {
+            buttonChooseFile =
+                    new JButton(Constant.messages.getString(MESSAGE_PREFIX + "chooseFileButton"));
+            buttonChooseFile.addActionListener(
+                    e -> {
+                        JFileChooser filechooser =
+                                new JFileChooser(
+                                        Model.getSingleton().getOptionsParam().getUserDirectory());
+                        int state = filechooser.showOpenDialog(View.getSingleton().getMainFrame());
+                        if (state == JFileChooser.APPROVE_OPTION) {
+                            String filename = filechooser.getSelectedFile().getAbsolutePath();
+                            try {
+                                getSchemaField().setText(filename);
+                                Model.getSingleton()
+                                        .getOptionsParam()
+                                        .setUserDirectory(filechooser.getCurrentDirectory());
+                            } catch (Exception e1) {
+                                showWarningDialog(
+                                        Constant.messages.getString(
+                                                "graphql.error.filenotfound", filename));
+                            }
+                        }
+                    });
+        }
+        return buttonChooseFile;
+    }
+
+    public boolean importDefinition() {
```

Comment:
I think the accessibility of the methods should be reviewed now, to be stricter.